contrib/export2collection.lua

```diff
@@ -0,0 +1,192 @@
+  result["path"], result["filename"], result["basename"], result["filetype"] = string.match(str, "(.-)(([^\\/]-)%.?([^%.\\/]*))$")
+  return result
+end
+
+local function get_path(str)
+  local parts = split_filepath(str)
+  return parts["path"]
+end
+
+local function get_filename(str)
+  local parts = split_filepath(str)
+  return parts["filename"]
+end
+
+local function get_basename(str)
+  local parts = split_filepath(str)
+  return parts["basename"]
+end
+
+local function get_filetype(str)
+  local parts = split_filepath(str)
+  return parts["filetype"]
+end
+
+function check_if_bin_exists(bin)
+  local result = os.execute("which " .. bin)
+  if not result then
+    result = false
+  end
+  return result
+end
+
+-- Thanks Tobias Jakobs for the idea 
+function check_if_file_exists(filepath)
+  local result = os.execute("test -e " .. filepath)
+  if not result then
+    result = false
+  end
+  return result
+end
+
+local function filename_increment(filepath)
+
+  -- break up the filepath into parts
+  local path = get_path(filepath)
+  local basename = get_basename(filepath)
+  local filetype = get_filetype(filepath)
+
+  -- check to see if we've incremented before
+  local increment = string.match(basename, "_(%d-)$")
+
+  if increment then
+    -- we do 2 digit increments so make sure we didn't grab part of the filename
+    if string.len(increment) > 2 then
+      -- we got the filename so set the increment to 01
+      increment = "01"
+    else
+      increment = string.format("%02d", tonumber(increment) + 1)
+      basename = string.gsub(basename, "_(%d-)$", "")
+    end
+  else
+    increment = "01"
+  end
+  local incremented_filepath = path .. basename .. "_" .. increment .. "." .. filetype
+
+  dt.print_error("original file was " .. filepath)
+  dt.print_error("incremented file is " .. incremented_filepath)
+
+  return incremented_filepath
+end
+
+local function show_status(storage, image, format, filename,
+  number, total, high_quality, extra_data)
+    dt.print(string.format(_("Export Image %i/%i"), number, total))
+end
+
+local function file_move(fromFile, toFile)
+  local success = os.rename(fromFile, toFile)
+  if not success then
+    -- an error occurred, so let's try using the operating system function
+    if checkIfBinExists("mv") then
+      success = os.execute("mv '" .. fromFile .. "' '" .. toFile .. "'")
+    end
+  end
+  return success  -- nil on error, some value if success
+end
+
+local function import_exports(storage, image_table, extra_data) --finalize
+
+  -- for each of the image, exported image pairs
+  --   move the exported image into the directory with the original  
+  --   then import the image into the database which will group it with the original
+  --   and then copy over any tags other than darktable tags
+
+  for image,exported_image in pairs(image_table) do
+
+    local myimage_name = image.path .. "/" .. get_filename(exported_image)
+
+    while check_if_file_exists(myimage_name) do
```

Comment:
this loop could be made into a utility function find_unique_file_name or something like that (in dtutils.file)